stdlib/public/runtime/MetadataLookup.cpp

```diff
@@ -639,14 +639,16 @@ Optional<unsigned> findAssociatedTypeByName(const ProtocolDescriptor *protocol,
   unsigned matchingAssocTypeIdx = 0;
   bool found = false;
   while (!associatedTypeNames.empty()) {
-    auto split = associatedTypeNames.split(' ');
-    if (split.first == name) {
+    // Avoid using StringRef::split because its definition is not
+    // provided in the header so that it requires linking with libSupport.a.
+    auto splitIdx = associatedTypeNames.find(' ');
+    if (associatedTypeNames.substr(0, splitIdx) == name) {
       found = true;
       break;
     }
 
     ++matchingAssocTypeIdx;
-    associatedTypeNames = split.second;
+    associatedTypeNames = associatedTypeNames.substr(splitIdx).substr(1);
```

Comment:
Why not just do `associatedTypeNames.substr(splitIdx + 1)`?

Reply:
If the find() fails, it returns StringRef::npos. substr(npos) is defined to return an empty string, and StringRef().substr(1) also produces an empty string -- matching the original behavior with split(). If you do substr(npos+1) it will overflow and do the wrong thing.

Reply:
Ok. Makes sense.